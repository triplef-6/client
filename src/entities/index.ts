export * from "./tour"
export * from "./location"
export * from "./review"
export * from "./user"
export * from "./tags"