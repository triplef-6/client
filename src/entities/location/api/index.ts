export {getLocations} from "./getLocations.ts"
