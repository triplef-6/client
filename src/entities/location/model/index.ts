export {useLocations} from "./useLocations.ts"