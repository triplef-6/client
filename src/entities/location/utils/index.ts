export {sortLocations} from "./sortLocations.ts"