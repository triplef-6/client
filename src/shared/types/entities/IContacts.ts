export interface IContacts {
    vk?: string
    telegram?: string
    phone: string
}