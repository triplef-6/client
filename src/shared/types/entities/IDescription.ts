export interface IDescription {
    info: string
    whatToExpect: string
    places: string[]
    topics: string[]
    orgDetails: string
    meetingPlace: string
}