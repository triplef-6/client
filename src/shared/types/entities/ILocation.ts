export interface ILocation {
    id: number
    city: string
    country: string
    tourCount: number
    region: string
    image?: string
}