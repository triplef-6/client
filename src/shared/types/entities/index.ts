export * from "./ITour.ts"
export * from "./IDescription.ts"
export * from "./ILocation.ts"
export * from "./IMe.ts"
export * from "./IContacts.ts"
export * from "./IReview.ts"
export * from "./IOrder.ts"
export * from "./IUser.ts"