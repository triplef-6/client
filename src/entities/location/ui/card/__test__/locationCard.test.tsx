import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import {LocationCard} from "@/entities/location";

describe("Location card", () => {

    const mockProps = {
        country: "Россия",
        city: "Омск",
        tourCount: 52,
        image: "https://example.com/omsk.webp",
    }

    it("Проверка рендеринга компонента", () => {

        render(<LocationCard {...mockProps} />)

        expect(screen.getByText(mockProps.country)).toBeInTheDocument()
        expect(screen.getByText(mockProps.city)).toBeInTheDocument()
        expect(screen.getByText(`${mockProps.tourCount}+ экскурсий`)).toBeInTheDocument()

    })

    it("Проверка того, что переданный фон устанавливается", () => {
        render(<LocationCard {...mockProps} />)
        expect(screen.getByRole("locationCardBackground")).toHaveStyle(`background-image: url(${mockProps.image})`)
    })

    it("Скриншотный тест компонента с переданными пропсами", () => {
        const { container } = render(<LocationCard {...mockProps} />)
        expect(container).toMatchSnapshot()
    })

})