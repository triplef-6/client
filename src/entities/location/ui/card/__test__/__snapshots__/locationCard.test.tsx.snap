// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Location card > Скриншотный тест компонента с переданными пропсами 1`] = `
<div>
  <div
    class="_layout_802c16 _sizes_802c16 _bg_802c16 _transitions_802c16 group"
    role="locationCardBackground"
    style="background-image: url(https://example.com/omsk.webp);"
  >
    <div
      class="_bg_459f4f _opacity_459f4f"
    />
    <div
      class="_layout_ecf88f"
    >
      <span
        class="_country_ecf88f"
      >
        Россия
      </span>
      <span
        class="_city_ecf88f"
      >
        Омск
      </span>
      <span
        class="_count_ecf88f"
      >
        52
        + экскурсий
      </span>
    </div>
  </div>
</div>
`;
